test/Constraints/tuple_arguments.swift

```diff
@@ -1728,7 +1728,7 @@ do {
 do {
   func f(_: Int...) {}
   let _ = [(1, 2, 3)].map(f) // expected-error {{no exact matches in call to instance method 'map'}}
-  // expected-note@-1 {{found candidate with type '(((Int, Int, Int)) -> _) -> Array<_>'}}
+  // expected-note@-1 2{{found candidate with type '(((Int, Int, Int)) -> T) -> [T]'}}
```

Comment:
We now get 2 notes here because `T` is a generic parameter from 2 different decls. We ought to be able to fix this by canonicalizing ErrorType's original type when computing the canonical type, I can look into that in a follow-up